### PR TITLE
RoleStore migration

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

Migration of `RoleStore` to return plain errors.

Fixes https://github.com/mattermost/mattermost-server/issues/15034

### CODE REVIEW

**Comment by rvillablanca** [comment]
@agnivade  CI is failing but I could't reproduce the test, the mine is passing ok :man_shrugging:. Tomorrow I will investigate more

File: store/sqlstore/role_store.go [hunk 1]
```diff
@@ -149,27 +148,24 @@ func (s *SqlRoleStore) createRole(role *model.Role, transaction *gorp.Transactio
...
-func (s *SqlRoleStore) GetAll() ([]*model.Role, *model.AppError) {
+func (s *SqlRoleStore) GetAll() ([]*model.Role, error) {
 	var dbRoles []Role
 
 	if _, err := s.GetReplica().Select(&dbRoles, "SELECT * from Roles", map[string]interface{}{}); err != nil {
-		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlRoleStore.GetAll", "store.sql_role.get_all.app_error", nil, err.Error(), http.StatusNotFound)
-		}
-		return nil, model.NewAppError("SqlRoleStore.GetAll", "store.sql_role.get_all.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "failed to find Roles")
```
Comment (agnivade) [inline store/sqlstore/role_store.go:168]: Aren't we supposed to return a store.ErrNotFound here?

Comment (rvillablanca) [reply store/sqlstore/role_store.go:168]: Afaik Select doesn't return ErrNoRows

Comment (agnivade) [reply store/sqlstore/role_store.go:168]: Ah that's right. Thanks.

**Comment by srkgupta** [comment]
@jespino or @agnivade 
Can any of you please let me know what pages are affected because of this PR?

**Comment by jespino** [comment]
@srkgupta this affects the whole application from the permissions perspective, but specially, talking about pages, probably it affects custom permission schemes. Anyway, is not something that is too exposed to the users, is more an internal thing.

**Comment by srkgupta** [comment]
@agnivade or @jespino 
Can you please trigger the EE build for this. Its required for me to run the cypress tests on the CI.

**Comment by rvillablanca** [comment]
Thanks @srkgupta

**Pull request merged**